.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -20,6 +20,12 @@ See [DEVELOPMENT.md](https://github.com/aws-deadline/deadline-cloud/blob/mainlin
 - Are relevant docstrings in the code base updated?
 - Has the README.md been updated? If you modified CLI arguments, for instance.
 
+### Does this PR introduce new dependencies?
+
+- Does the PR add new package dependencies to `dependencies` section of `pyproject.toml`?
```

Comment:
A short description of why it's better to minimize dependencies could help with reasoning about it. The main reason is that this library is designed to be integrated into third-party applications by customers who have highly customized deployments. Every additional dependency increases the risk of version conflicts with something they've already set up.

Reply:
Nice! I'll add a framing into here.